ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java

```diff
@@ -69,6 +69,21 @@ public void setTextAlign(ReactTextView view, @Nullable String textAlign) {
       throw new JSApplicationIllegalArgumentException("Invalid textAlign: " + textAlign);
     }
   }
+  
+  @ReactProp(name = "textDecorationLine")
+    public void setTextDecorationLine(ReactTextView view, @Nullable String textDecorationLine) {
+        if (textDecorationLine == null || "none".equals(textDecorationLine)) {
+            view.getPaint().setFlags(0);
+        } else if ("underline".equals(textDecorationLine)) {
+            view.getPaint().setFlags(Paint.UNDERLINE_TEXT_FLAG | Paint.ANTI_ALIAS_FLAG);
+        } else if ("line-through".equals(textDecorationLine)) {
+            view.getPaint().setFlags(Paint.STRIKE_THRU_TEXT_FLAG | Paint.ANTI_ALIAS_FLAG);
+        } else if ("underline line-through".equals(textDecorationLine)) {
```

Comment:
@astreet there is a PR (incomplete) to support flexible parsing: #3819. So either that PR should get updated or this PR should borrow code from the other one.